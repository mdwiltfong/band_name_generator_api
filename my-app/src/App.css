.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-button {

  align-items: center;
  background-image: linear-gradient(144deg,#AF40FF, #5B42F3 50%,#00DDEB);
  border: 0;
  border-radius: 8px;
  box-shadow: rgba(151, 65, 252, 0.2) 0 15px 30px -5px;
  box-sizing: border-box;
  color: #FFFFFF;
  display: flex;
  font-family: Phantomsans, sans-serif;
  font-size: 20px;
  justify-content: center;
  line-height: 1em;
  max-width: 100%;
  min-width: 140px;
  padding: 19px 24px;
  margin-bottom: 50px;
  text-decoration: none;
  user-select: none;
  -webkit-user-select: none;
  touch-action: manipulation;
  white-space: nowrap;
  cursor: pointer
}

.button-63:active,
.button-63:hover {
  outline: 0;
}

@media (min-width: 768px) {
  .button-63 {
    font-size: 24px;
    min-width: 196px;
  }
}
  
  
  .button-1:hover,
  .button-1:focus {
    background-color: #F082AC;
  }

  .App-message {
    font-size: 24px;
    min-height: 53px;
    display: flex;
    flex-direction: column;
  }

  .App-message button{
    max-width: 100px;
    margin: 0 auto;
  }

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}