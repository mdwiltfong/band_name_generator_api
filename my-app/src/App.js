import "./App.css";
import BandName from "./components/band_name/BandName";
import BandList from "./components/band_list/BandList";
import NavBar from "./components/sign_up/NavBar";
import { Outlet } from "react-router-dom";

const App = () => {
  return (
    <>
      <div className="App">
        <header className="App-header">
          <NavBar />
          <BandName />
          <BandList />
        </header>
      </div>
      <div id="user_sign_up">
        <Outlet />
      </div>
    </>
  );
};

export default App;
