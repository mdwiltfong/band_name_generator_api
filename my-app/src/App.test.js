import { fireEvent, render, waitFor } from "@testing-library/react";
import { scryRenderedComponentsWithType } from "react-dom/test-utils";
import App from "./App";

describe("App component unit tests", () => {
  beforeEach(() => {
    /* TODO: Add dbReset function to these tests to preserve db */
  });
  it("Smoke Test", () => {
    render(<App />);
  });
  it("Snapshot tests", () => {
    const screen = render(<App />);
    expect(screen.asFragment()).toMatchSnapshot();
  });
  it("should generate random bandname when button is clicked", async () => {
    const screen = render(<App />);
    const generateBtn = screen.getByTestId("generate-btn");
    fireEvent.click(generateBtn);

    await waitFor(() => {
      screen.debug();
      const addBtn = screen.getByTestId("add-band-btn");
      screen.debug();
      expect(addBtn).toBeInTheDocument();
    });
  });
  it("a list of bandnames should appear when list-bandnames-btn is clicked", () => {
    const screen = render(<App />);
    const listBtn = screen.getByText("See a list of upvoted band names!");
    fireEvent.click(listBtn);
    const bandList = screen.container.getElementsByClassName("band");
    screen.debug();
    expect(bandList.length).not.toBe(null);
  });
  it.todo("User can add bandname to database");
  /* Click on "What should we call the band" Is there a Add band button? When we click it, does it 
  // show the confirmation message? Can it show the error message? 
  Does the new bandname hit the BandResult component? 
  
  */
});
