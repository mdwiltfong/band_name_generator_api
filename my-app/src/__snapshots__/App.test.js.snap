// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component unit tests Snapshot tests 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <button
        class="App-button"
        data-testid="generate-btn"
        id="button"
        type="button"
      >
        What should we call the band?
      </button>
      <div
        class="band-list"
      >
        <button
          class="App-button"
          id="button"
          type="button"
        >
           See a list of upvoted band names!
        </button>
        <div
          class="bands"
        />
      </div>
    </header>
  </div>
</DocumentFragment>
`;
